tests/ptvsd/common/test_launcher.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os.path
+import platform
+import pytest
+import subprocess
+import sys
+
+from ptvsd.common import launcher
+
+
+launcher_py = os.path.abspath(launcher.__file__)
+
+
+@pytest.mark.parametrize("run_as", ["file", "module", "code"])
+@pytest.mark.parametrize("mode", ["normal", "abnormal", "normal+abnormal", ""])
+@pytest.mark.parametrize("seperator", ["seperator", ""])
+def test_launcher_parser(mode, seperator, run_as):
+    args = []
+
+    switch = mode.split("+") if "+" in mode else [mode]
```

Comment:
No need to check - you can just split, and you'll get back a single-item list if the separator was not there.